amgut/lib/data_access/ag_data_access.py

```diff
@@ -142,14 +107,14 @@ def check_login_exists(self, email):
         ag_login_id or None
             If exists, returns ag_login_id, else returns None
         """
-        clean_email = email.strip().lower()
-        sql = "select ag_login_id from ag_login WHERE LOWER(email) = %s"
-        cur = self.get_cursor()
-        cur.execute(sql, [clean_email])
-        value = cur.fetchone()
-        if value is not None:
-            value = value[0]
-        return value
+        with TRN:
+            clean_email = email.strip().lower()
+            sql = "SELECT ag_login_id FROM ag_login WHERE LOWER(email) = %s"
+            TRN.add(sql, [clean_email])
+            value = TRN.execute_fetchindex()
+            if not value:
```

Comment:
oh, I see, it's a list. Could you invert the logic? less thought needed... and then the `return None` is not necessary